testsuite/tests/ui/mail_and_messages/test_automatic_mails.py

```diff
@@ -2,12 +2,35 @@
 import re
 import pytest
 
-
+from testsuite import rawobj
 from testsuite.ui.views.admin.audience.account import AccountInvitationNewView
+from testsuite.ui.views.admin.audience.application import ApplicationDetailView
+from testsuite.ui.views.devel.applications import DevelApplicationDetailView
 from testsuite.ui.views.devel.login import InvitationSignupView
 from testsuite.utils import blame
 
 
+@pytest.fixture(scope="module")
+def extra_plan(request, service, custom_app_plan):
+    """Additional application plan to choose"""
+    return custom_app_plan(rawobj.ApplicationPlan(blame(request, "extra_plan")), service)
+
+
+@pytest.fixture(scope="module")
+def custom_app(service, custom_app_plan, custom_application, lifecycle_hooks, request):
+    """
+    Create an application with predefined application plans
```

Comment:
This is insufficient comment (many other are, different people, different PRs, but here it is more important). This doesn't explain why it is needed and how it is used. Better would be that application is created during test itself and therefore this is needed.

Anyway I'd rather suggest different approach. To plans, one for each test, and one app (main application fixture) sound better to me. On the other hand this proposed solution would work if upgrade testing using pytest_persistence would be in use, but a model with two plans would not. However that is more a problem of pytest_persistence and less of the design.

That being written I am not requesting the change to two plans, that's something to think about. I am requesting the change of docstring with no doubt.

Reply:
Finally, I remember why I do it this way and I absolutely agree with you that your proposed flow makes much more sense.  But, the implementation of a developer portal is not optimal, and when a single application has more than two plans active we would need to extend some Widgets locators and usage of some of our base elements because each plan has its own buttons that are not generated dynamically, but only overlayed one by another. I will try to change it to your proposed solution.